### DISCUSSION TITLE
Convert nullable type to class

### DISCUSSION THREAD

**Discussion by marcvoor** [LuckyPennySoftware/AutoMapper#4222] [NA]
I have a very simple class, wrapping a double and providing me a custom ToString():
```
public class Age {
  public double Value { get; set; }
  public Age() {}
  public Age(double age) => Value = age;
  public override string ToString() => $"{Value} years";
}
```
With Automapper (version 10.1.1) I try to map a `double?` to this `Age` class, but whatever I try, it won't work. See below my tests.

My first try was the 'plain' way:
```
var config = new MapperConfiguration(cfg => {
   cfg.CreateMap<double?, Age>();
});
var plan = config.BuildExecutionPlan(typeof(double?), typeof(Age));
var mapper = config.CreateMapper();

double? value = null;
var age = mapper.Map<Age>(value);
Assert.AreEqual(0, age);
```
I do understand that this does not work, because when I check the execution plan, it clearly shows the null check done by AutoMapper.

So, my next try was a custom TypeConverter:
```
public class CustomAgeResolver : ITypeConverter<double?, Age>
{
   public Age Convert(double? source, Age destination, ResolutionContext context)
   {
      return new Age(source.GetValueOrDefault());
   }
}

var config = new MapperConfiguration(cfg => {
   cfg.CreateMap<double?, Age>().ConvertUsing<CustomAgeResolver>();
});
var plan = config.BuildExecutionPlan(typeof(double?), typeof(Age));
var mapper = config.CreateMapper();

double? value = null;
var age = mapper.Map<Age>(value);
Assert.AreEqual(0, age);
```

This solution does not work either. Checking the execution plan tells me that the `Convert` method of the `CustomAgeResolver` should be called, but that does not happen.

Am I doing something wrong here? Or is there an easier way to implement the AutoMapping?
Any advice is welcome!

**Comment by lbargaoanu** [comment]
`mapper.Map<double?, Age>(value)`.

**Reply by marcvoor** [reply]
Thanks, that helps!
